app/controllers/episodes_controller.rb

```diff
@@ -1,11 +1,48 @@
 class EpisodesController < ApplicationController
   load_and_authorize_resource
+  before_action :set_episode, only: %i[edit show]
 
   def index
     @episodes = Episode.order(created_at: :desc).page params[:page]
   end
 
-  def show
+  def show; end
+
+  def edit; end
+
+  def create
+    @episode = Episode.new(episode_params.merge(created_by: current_user))
+
+    if @episode.save
+      redirect_to episodes_url, notice: 'Successfully created'
+    else
+      render :new, notice: 'Something wrong'
+    end
+  end
+
+  def update
+    if Episode.find(params[:id]).update(episode_params)
+      redirect_to episode_path
+    else
+      redirect_to edit_episode_path
+    end
+  end
+
+  private
+
+  def set_episode
     @episode = Episode.find(params[:id])
   end
+
+  def episode_params
+    params[:video] = params[:video].split('=').last
```

Comment:
? Why we make it here?

Reply:
Because we prepare parameters in this method before saving and updating.

Reply:
You shouldn't modify params object, first permit necessary params then from another method (should modify params) return hash with modified permitted params

Reply:
Also try for `video` method `match`

Reply:
> Also try for `video` method `match`

OK, but i really don't like this way.

Reply:
> You shouldn't modify params object, first permit necessary params then from another method (should modify params) return hash with modified permitted params

After this code looks bad